Packs/SymantecEndpointSecurity/Integrations/SymantecEndpointSecurity/SymantecEndpointSecurity.py

```diff
@@ -0,0 +1,481 @@
+        f"Total number of events after merging with last fetch events: {len(filtered_events)}"
+    )
+
+    try:
+        push_events(filtered_events)
+    except Exception as e:
+        # If the push of events to XSIAM fails,
+        # the current fetch's events are stored in `integration_context`,
+        # ensuring they are pushed in the next fetch operation.
+        calculate_next_fetch(
+            filtered_events=filtered_events,
+            next_hash=next_hash,
+            include_last_fetch_events=True,
+            last_integration_context=integration_context,
+        )
+        raise DemistoException(
+            "Failed to push events to XSIAM, The integration_context updated"
+        ) from e
+
+    calculate_next_fetch(
+        filtered_events=filtered_events,
+        next_hash=next_hash,
+        include_last_fetch_events=False,
+        last_integration_context=integration_context,
+    )
+
+
+def perform_long_running_loop(client: Client):
+    """
+    Manages the fetch process.
+    Due to a limitation on Symantec's side,
+    the integration is configured as long-running
+    since API calls can take over 5 minutes.
+
+    Fetch process:
+        - In every iteration except the first,
+          fetch is performed with the `next_fetch` argument,
+          which acts as a pointer for Symantec.
+        - When an error is encountered from Symantec,
+          it is handled based on the error type, and before the next iteration,
+          the process enters a brief sleep period defined by `FETCH_INTERVAL`
+          to avoid overloading with API calls.
+    """
+    while True:
+        # Used to calculate the duration of the fetch run.
+        start_timestamp = time.time()
+        try:
+            integration_context = get_integration_context()
+            demisto.info(f"Starting new fetch with {integration_context=}")
+            get_events_command(client, integration_context=integration_context)
+
+        except UnauthorizedToken:
+            try:
+                client._update_access_token()
+            except Exception as e:
+                raise DemistoException("Failed obtaining a new access token") from e
+        except NextPointingNotAvailable:
+
+            demisto.debug(
+                "Next is pointing to older event which is not available for streaming. "
+                "Clearing next_fetch, The integration's dedup mechanism will make sure we don't insert duplicate events. "
+                "We will eventually get a different pointer and fetching will overcome this edge case"
+            )
+            integration_context.pop("next_fetch")
+            set_integration_context(integration_context)
+        except Exception as e:
+            raise DemistoException("Failed to fetch logs from API") from e
+
+        # Used to calculate the duration of the fetch run.
+        end_timestamp = time.time()
+
+        sleep_if_necessary(client, start_timestamp, end_timestamp)
```

Comment:
passing timestamps in seconds may be confusing. 
The alternative of passing two `datetime` objects, then deducting them in seconds may be confusing as well, so please change `sleep_if_necessary` to accept a client and the `last_run_duration`, being result of deduction the two timestamps. 

```suggestion
        sleep_if_necessary(client, end_timestamp - start_timestamp)
```